### PR TITLE
Add FlattenGraph for GraphNode

### PR DESCRIPTION
**Pull request by omerzi**
- [ ] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: xray/services/scan.go [hunk 1]
```diff
@@ -189,6 +190,34 @@ type GraphNode struct {
...
+func FlattenGraph(graph []*GraphNode) []*GraphNode {
+	flatGraph := []*GraphNode{}
+	depsSet := datastructures.MakeSet[string]()
+	for _, node := range graph {
+		flatGraph = append(flatGraph, getUniqueDependencies(node, depsSet)...)
+	}
+	return []*GraphNode{{Id: "root", Nodes: flatGraph}}
+}
+
+func getUniqueDependencies(node *GraphNode, depsSet *datastructures.Set[string]) []*GraphNode {
```
Comment (yahavi) [inline xray/services/scan.go:203]: Wouldn't work better with a map of string -> *GraphNode?
The key is the dependency ID and the value is the GraphNode. No return value.

Possible implementation (please double check me):
```go
func FlattenGraph(graph []*GraphNode) []*GraphNode {
	allDependencies := map[string]*GraphNode{}
	for _, node := range graph {
		populateUniqueDependencies(node, allDependencies)
	}
	return []*GraphNode{{Id: "root", Nodes: maps.Values(allDependencies)}}
}

func populateUniqueDependencies(node *GraphNode, allDependencies map[string]*GraphNode) {
	if _, exist := allDependencies[node.Id]; exist {
		return
	}
	allDependencies[node.Id] = &GraphNode{Id: node.Id}
	for _, dependency := range node.Nodes {
		populateUniqueDependencies(dependency, allDependencies)
	}
}
```

File: xray/services/scan_test.go [hunk 2]
```diff
@@ -47,3 +49,32 @@ func TestCreateScanGraphQueryParams(t *testing.T) {
 		})
 	}
 }
+
+func TestFlattenGraph(t *testing.T) {
```
Comment (yahavi) [inline xray/services/scan_test.go:53]: Let's add another test to check no duplications

**Review by yahavi — CHANGES_REQUESTED** [review]

File: xray/services/scan.go (hunk 1 above)
Comment (omerzi) [reply xray/services/scan.go:203]: Nice! Didn't know about the `maps.Values()`. Works great.

File: xray/services/scan_test.go (hunk 2 above)
Comment (omerzi) [reply xray/services/scan_test.go:53]: Already checked it in the test (with the set)

**Review by eyalbe4 — APPROVED** [review]

**Pull request merged**